### PR TITLE
Inform Readers Of 10 Minute Tutorial Of Prerequisites 

### LINKED ISSUE #647: Add a note in the introduction of the 10 minutes tutorial to inform that it prerequisites basic development knowledge
The "10 min tutorials" on https://docs.cucumber.io/docs/guides/10-minute-tutorial/ are frustrating to use. It presume much from the readers when it suddenly swift between Command-promt windows and Ruby etc. between steps. It presumes that that the reader would understand "Gemfile", let alone how to use them.

This may be enough information for someone who already know Ruby and Cucumber from the get-go (which would make a tutorial unnecessary by definition), but when you're a new user who only know "enough" to get around, but not to get into every crevasse, then it's a frustrating experience. It feels at times as if you get talk down to for them not knowing what the tutorial, by definition, should have been able to tell me in the first place.

**Describe the solution you'd like**
Images/Screenshots that shows when and where you are at any given point. To show where you as a reader are when the tutorial-text jump around in the menu. To show which window/command window etc. you're currently in at any given moment.

Describe alternatives you've considered
Bulleting, that clearly states the How and Where to do the how.

Additional context
(See image for reference).
That is just an example of the "installing process". There are plenty more in the text, but for this example, you can see the attatched screenshot for reference.

1: It start with "We’ll start by creating a new directory and an empty Ruby project." (In the same project or as two separate entities? Is the directory specific to the project or a new default? Should the directory be called something specific (like Gemfile) or is that optional?)

Then it goes down to "Create a Gemfile with the following content:" - As a new user and as a ready, the question on ones mind is "Where?" In the directory? In the project file itself? In a new place? Should the Gemfile be given a specific filetype? (like Gemfile.rb?) Should the reader just write-in the versions as they are presented on the sire (Screenshot), or should they use the command-promt to find the right versions? (cucumber -version or other similar?). Does it matter?

Thirdly, we get to "Install Cucumber and prepare the file structure:" - Again, is that in Ruby or in command promt? Is the reader still in the Gemfile? Should this be added at the start of a new [File].feature?

I hope that this shows the issue I personally have had as a new user. The tutorial jump left and right with little explanation or consideration to anyone who supposedly not already possesses enough knowledge to make a tutorial void in the first place. It is called a "10 minutes tutorial", but personally, I have yet to reach the end of the tutorial. This is frustrating, partly due to the aforementioned "jumps", but also when you do manage to follow the tutorial, since when you come across an issue that halt one process, then you don't know why. This is especially an issue when it comes to writing an issue-report or asking for help, because you cannot be sure that you installed things right in the first place.

text ex

### PR DESCRIPTION
**Pull request by TomerPacific**
Adding introduction bullets for java/kotlin/javascript/ruby

# Description

In order to better clarify that the 10 minute guides need some prior knowledge, several bullet points were added per language to demonstrate this.

# Motivation & context
Fixes #647 

## Type of change
- New feature (non-breaking change which adds new behaviour)

## Note to other contributors
N/A

# Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING**](../CONTRIBUTING.md) document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by TomerPacific** [comment]
Once we agree upon the format for these bullets, there might be a need to consolidate the Java and Kotlin cases together.

File: content/docs/guides/10-minute-tutorial.md [hunk 1]
```diff
@@ -21,6 +21,40 @@ In this quick tutorial you will learn how to:
...
 Friday yet.
 
+Please be aware that this tutorial assumes that you have a:
+
+{{% block "java" %}}
+
+* Basic understanding of the Java programming language
+* Basic understanding of the Gradle file
+
+{{% block %}}
```
Comment (aurelien-reeves) [inline content/docs/guides/10-minute-tutorial.md]:
```suggestion
{{% /block %}}
```

File: content/docs/guides/10-minute-tutorial.md [hunk 2]
```diff
@@ -21,6 +21,40 @@ In this quick tutorial you will learn how to:
...
+
+{{% block "kotlin" %}}
+
+* Basic understanding of the Kotlin programming language
+* Basic understanding of the Gradle file
+
+{{% block %}}
```
Comment (aurelien-reeves) [inline content/docs/guides/10-minute-tutorial.md]:
```suggestion
{{% /block %}}
```

File: content/docs/guides/10-minute-tutorial.md [hunk 3]
```diff
@@ -21,6 +21,40 @@ In this quick tutorial you will learn how to:
...
+
+{{% block "javascript" %}}
+
+* Basic understanding of the Javascript programming language
+
+{{% block %}}
```
Comment (aurelien-reeves) [inline content/docs/guides/10-minute-tutorial.md]:
```suggestion
{{% /block %}}
```

File: content/docs/guides/10-minute-tutorial.md [hunk 4]
```diff
@@ -21,6 +21,40 @@ In this quick tutorial you will learn how to:
...
+
+{{% block "ruby" %}}
+
+* Basic understanding of the Ruby programming language
+
+* Basic understanding of the Bundler tool and `Gemfile`
+
+{{% block %}}
```
Comment (aurelien-reeves) [inline content/docs/guides/10-minute-tutorial.md]:
```suggestion
{{% /block %}}
```

**Review by aurelien-reeves — CHANGES_REQUESTED** [review]
The ending blocks are not ending 😄

But regarding the format and the content, it looks good to me 👍

File: content/docs/guides/10-minute-tutorial.md (hunk 4 above)
Comment (TomerPacific) [reply content/docs/guides/10-minute-tutorial.md]: Have corrected ending blocks.

File: content/docs/guides/10-minute-tutorial.md [hunk 5]
```diff
@@ -21,6 +21,40 @@ In this quick tutorial you will learn how to:
...
+* Basic understanding of the Ruby programming language
+
+* Basic understanding of the Bundler tool and `Gemfile`
+
+{{% /block %}}
+
+* Some experience using a terminal
+
+* Some experience using a text editor
+
```
Comment (aurelien-reeves) [inline content/docs/guides/10-minute-tutorial.md]:
```suggestion
Please be aware that this tutorial assumes that you have a:

{{% block "java" %}}
* Basic understanding of the Java programming language
* Basic understanding of the Gradle file
{{% /block %}}
{{% block "kotlin" %}}
* Basic understanding of the Kotlin programming language
* Basic understanding of the Gradle file
{{% /block %}}
{{% block "javascript" %}}
* Basic understanding of the Javascript programming language
{{% /block %}}
{{% block "ruby" %}}
* Basic understanding of the Ruby programming language
* Basic understanding of the Bundler tool and `Gemfile`
{{% /block %}}
* Some experience using a terminal
* Some experience using a text editor

```
Blank lines in a list are generating extra paragraph and new lists in the resulting HTML

**Review by aurelien-reeves — CHANGES_REQUESTED** [review]

**Comment by aurelien-reeves** [comment]
FYI in the "checks" list, if you click the "Details" link for "netlify/cucumber/deploy-preview", you can see a preview of your changes. You'll eventually just have to put "/docs" manually at the end of the URL.

**Comment by TomerPacific** [comment]
@aurelien-reeves , your proposed changes unformatted the lists and they appear as *. Working on sorting it out.

docs

**Comment by aurelien-reeves** [comment]
Yeap, just noticed that too.

I guess maybe just adding a blank line right before the "some experience using a terminal" might fix this

**Review by aurelien-reeves — APPROVED** [review]

**Comment by TomerPacific** [comment]
@aurelien-reeves , thank you for your patience and for reviewing this PR. What is the process now after this PR has been approved? Should I merge it?

**Comment by aurelien-reeves** [comment]
Yes, you can merge
It will be deployed automatically to production

**Comment by aurelien-reeves** [comment]
Thanks @TomerPacific for your contribution 🎉

**Pull request merged**